clients/drcachesim/tests/memref_gen.h

```diff
@@ -84,7 +84,28 @@ gen_branch_encoded(memref_tid_t tid, addr_t pc, int encoding)
     memref.instr.encoding_is_new = true;
     return memref;
 }
+
+inline memref_t
+gen_syscall_encoded(memref_tid_t tid, addr_t pc, int encoding)
```

Comment:
There's nothing here specific to a syscall so the name should perhaps be gen_instr_encoded; for that matter maybe it should take in a trace entry type and we then eliminate the nearly-duplicate gen_branch_encoded and only have these ifdefs and different encoding types in one place.

Reply:
I suggested above to remove the encoding parameter let the routine add it on its own, which would make the routine specific to syscalls. Having a dedicated routine for syscalls avoids having to specify the encoding in each call. But I'm fine either way.

Reply:
I'm guessing in the future we may want to write instruction sequences with encodings for arbitrary code (I'm particularly thinking of testing virtual register dep chains) and won't want to create a "gen_add_with_such_operands" etc. for each variety so I would vote for not having a syscall-specific one (we would have encodings grabbed from IR and turned into a record via sthg like gen_instr_encoded).

Reply:
Removed the syscall specific one and added `gen_instr_encoded`.